frontend/src/pages/auth/register.page.vue

```diff
@@ -1,38 +1,151 @@
-    <input type="email" id="email" placeholder="Your Email" v-model.trim="email" />
-    <input type="password" id="password" placeholder="Your Password" v-model.trim="password" />
-    <button @click="onRegister">Register</button>
+    <base-card>
+      <h1>Register</h1>
+      <form @submit.prevent="onRegister">
+        <div class="form-control">
+          <input-validation
+            :withLabel="true"
+            label="Name"
+            :validator="nameValidator"
+            :empty="empty"
+            value="registerName"
+            placeholder="Your Name"
+          ></input-validation>
+          <input-validation
+            :withLabel="true"
+            label="Email"
+            :validator="emailValidator"
+            :empty="empty"
+            value="registerEmail"
+            type="email"
+            placeholder="Your Email"
+          ></input-validation>
+          <input-validation
+            :withLabel="true"
+            label="Password"
+            :validator="passwordValidator"
+            :empty="empty"
+            value="registerPassword"
+            type="password"
+            placeholder="Set Password"
+          ></input-validation>
+          <input-validation
+            :withLabel="true"
+            label="Confirm Password"
+            :validator="confirmPasswordValidator"
+            :empty="empty"
+            value="confirmPassword"
+            type="password"
+            placeholder="Confirm Password"
+          ></input-validation>
+        </div>
+        <link-button class="register-button">Register</link-button>
+        <link-button :to="Route.Login" type="button" :link="true" mode="flat" @click="clearInputs">
+          Have an account? Log in instead
+        </link-button>
+      </form>
+    </base-card>
   </Page>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 
 import Page from '@components/page/page.vue';
-import { authVuexModule } from '@src/vuex/store-accessor';
+import { authVuexModule, documentEditVuexModule } from '@src/vuex/store-accessor';
 import { CreateUserRequest } from '@shared/request-response';
+import { Route } from '@src/router/routes.enum';
+import BaseCard from '@components/cards/card.vue';
+import LinkButton from '@components/buttons/link-button.vue';
+import {
+  nameValidator,
+  emailValidator,
+  passwordValidator,
+  confirmPasswordValidator,
+} from '@src/validation/registerValidation';
+import InputValidation from '@components/inputs/input-validation.vue';
 
 export default defineComponent({
   components: {
     Page,
+    BaseCard,
+    LinkButton,
+    InputValidation,
   },
 
   data() {
     return {
-      name: '',
-      email: '',
-      password: '',
+      empty: false,
     };
   },
 
   methods: {
-    async onRegister() {
-      const { name, email, password } = this;
+    async onRegister(): Promise<void> {
+      const name = documentEditVuexModule.inputs.registerName;
```

Comment:
Why does onRegister use documentEditVuexModule?

Reply:
Has been fixed. No inputs are stored in vuex anymore